workflow.yaml

```diff
@@ -246,5 +260,75 @@ spec:
         env:
           - name: AGISOFT_FLS
             value: "{{workflow.parameters.AGISOFT_FLS}}"
+            
+  # --------- RCLONE UPLOAD (Docker image) ---------
+    - name: rclone-upload-to-s3
+      inputs:
+        parameters:
+          - name: dataset-name
+      container:
+        image: rclone/rclone:latest
+        volumeMounts:
+          - name: results
+            mountPath: /results
+        command: ["/bin/sh", "-lc"]
+        args:
+          - |
+            set -euo pipefail
+            SRC="/results/{{workflow.parameters.RUN_FOLDER}}/{{inputs.parameters.dataset-name}}"
+            DST=":s3:{{workflow.parameters.S3_BUCKET}}/{{workflow.parameters.RUN_FOLDER}}/{{inputs.parameters.dataset-name}}"
```

Comment:
In production I assume we'd want to drop the `RUN_FOLDER` from the final file tree, since we'd want it to be one flat structure for all datasets, no matter when they were processed. But this seems good for debugging now.